### ISSUE TITLE
The escaping of html file names in the dox system seems to be odd

### ISSUE THREAD

**Issue by rrahn** [seqan/seqan#831]
For example see ```FragmentStore::PredefinedAnnotationTypes```, where the frame-based link does not point to the correct address (when searching with ```?p=FragmentStore::PredefinedAnnotationTypes```).

The problem occurs, when the ```::``` in the name is converted in the ```write_html.py``` to some own encoding theme. Within ```common.js```, the URI is encoded using ```encodingURIComponent``` leading to a different sequence name. Moreover the script cuts of the part before the colons and directs it to the ```FragmentStore.html```, which is not the linked site to be displayed.

This needs some more thoughts and it would be sufficient to fix it after 2.0.0.
@holtgrewe @bkahlert Your insights to this would be nice!

**Comment by holtgrewe** [comment]
This is one of the things we inherited from Andrea's dddoc ;) Yes, let's talk about this stuff after 2.0

**Comment by h-2** [comment]
@xenigmax can you have a look? Thanks!

**Comment by rrahn** [comment]
Apparently the current solution does not work as it generates html files with invalid characters for windows platforms.
What we really need to do, is to replace the ```::``` part of the html file nam with the encoding scheme used by the write_html.py. 
But be cautious the ```::``` is also used to for displaying member functions which might cause some troubles to generically apply the encoding to all ```::``` occurrences.

**Comment by rrahn** [comment]
@xenigmax I wrote this little snippet here:
```
/**
 * Encode html file name. 
 * This function encodes html file names compliant with the write_html.py.
 */
(function ($) {
	$.extend({
		encodeName: function(name) {
            var tmp = "";
            for (var i = 0, len = name.length; i < len; i++) {           
                if (name[i].match(/[0-9a-z]/i))
                    tmp += name[i]
                else
                    tmp += '_' + name[i].charCodeAt(0);
            }
            // console.log('The encoded name: ' + tmp);
            return tmp;
		}
	});
})(jQuery);
```
We could use this!

**Comment by xenigmax** [comment]
@rrahn  
Hey.. I'm sorry but I'm really confused about those things..
I found another thing that does not work properly.

There are some cases which use "#" multiple times. Below is an example.
http://docs.seqan.de/seqan/master/?p=Fragment#Fragment%23fragmentBegin

Should we seprate this from the current one?

**Comment by xenigmax** [comment]
@rrahn 
In the .js code you suggested, we added "_" for skipping after discussion.

But there is a file which contains "_" in the middle of the name.
http://docs.seqan.de/seqan/master/macro_SEQAN_95HOST.html
So the .js should ignore first "_" only.

**Comment by xenigmax** [comment]
@rrahn  
All right. 
As far as I know, those 3 cases are not working.
There can be more but I can't figure out now.
I will fix all and will make a pull request.

**1. "::" in p**
example :
http://docs.seqan.de/seqan/master/?p=ArgParseArgument::ArgumentType
(not working in window system)

reason :
":" is not allowed in filename (windows)

**2. multiple "#"s**
example:
http://docs.seqan.de/seqan/master/?p=BamTagsDict#BamTagsDict%3A%3ABamTagsDict

reason:
The hash part is encoded(..%3A..), but NOT decoded before the redirection.

**3. "_" in the middle of the file name**
example:
http://docs.seqan.de/seqan/master/?p=SEQAN_HOST

reason:
Some html file contains "_" in the middle of the name.
And the python script encode this to "_95"

**Comment by xenigmax** [comment]
@rrahn 
Sadly, I found another one.

The html pages for macros encode "_" in the middle of the name to "_95".
http://docs.seqan.de/seqan/master/?p=SEQAN_HOST
filename : macro_SEQAN**_95**HOST.html

But, html pages for global function do not.
http://docs.seqan.de/seqan/master/?p=arrayFill
filename : global_function**_**arrayFill.html

I have to do something more~!

**Comment by xenigmax** [comment]
@rrahn 
I have fixed everything I mentioned above.
Let me know if you find another problem.

**Comment by rrahn** [comment]
Ok, this is all very messed up. I don't know if we will have some time in the future to improve this.
But your solutions seems to be quite good for the moment. 
Maybe we can split the naming scheme in general in: entity, page name and anchor so we can treat them without all this if-then clutter.
Thanks a lot.

**Comment by rrahn** [comment]
Fixed in #1414 and #1415
